src/types.js

```diff
@@ -13,6 +13,16 @@ export type Position = {|
   y: number,
 |};
 
+export type Direction = 'horizontal' | 'vertical';
+
+export type Axis = {|
```

Comment:
split into VerticalAxis and HorizontalAxis

type Axis = VerticalAxis | HorizontalAxis

no need to expose VerticalAxis or HorizontalAxis.

This prevents the case where an axis is made that has a mixture of vertical and horizontal properties